src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HPack/HPackEncoder.cs

```diff
@@ -128,68 +178,8 @@ private static bool EncodeStringPart(string s, Span<byte> buffer, ref int curren
                 }
                 buffer[i++] = (byte)c;
             }
-            currentIndex += i;
-
-            return true;
-        }
-
-        private static bool EncodeHeaderValue(string s, string[] parts, string separator, Span<byte> buffer, out int length)
-        {
-            int i = 0;
-            length = 0;
-
-            // \0, length and at least one byte of value
-            if (buffer.Length < 3)
-            {
-                return false;
-            }
-
-            if (s != null)
-            {
-                i = s.Length;
-            }
-            else
-            {
-                // Calculate length of all parts and separators.
-                foreach (string part in parts)
-                {
-                    i += part.Length;
-                }
-
-                i += (parts.Length - 1) * separator.Length;
-                s = parts[0];
-            }
-
-            if (!IntegerEncoder.Encode(i, 7, buffer, out int valueLength))
-            {
-                return false;
-            }
-
-            i = valueLength;
-
-            if (!EncodeStringPart(s, buffer.Slice(i), ref i))
-            {
-                return false;
-            }
+            encodedLength = i;
```

Comment:
It's probably better to use a for loop here instead of foreach.

Reply:
is there performance benefit or is it mostly style?

Reply:
I think there's a perf benefit, because we don't need to allocate the IEnumerator. But that may be optimized away, not sure.

Reply:
ok. that would make sense.

Reply:
> But that may be optimized away, not sure.

The C# compiler recognizes foreach loops over arrays, strings, and spans, and turns them into the equivalent of a for loop.  Even if it didn't, we do expose a struct-returning GetEnumerator on span to avoid that allocation, though per the previous sentence it'll rarely be used.